pil-analyzer/tests/condenser.rs

```diff
@@ -174,3 +174,95 @@ pub fn constructed_constraints() {
 "#;
     assert_eq!(formatted, expected);
 }
+
+#[test]
+#[should_panic = "There are pre-existing constraints or witness columns"]
+pub fn capture_stage_non_fresh_wit() {
+    let input = r#"
+        namespace std::prover;
+            let capture_stage: (-> int) -> Constr[] = 9;
+
+        namespace Main(1024);
+            col witness x;
+            let f = constr || { x = 2; 1024 };
+            std::prover::capture_stage(f);
+    "#;
+    analyze_string::<GoldilocksField>(input);
+}
+
+#[test]
+#[should_panic = "There are pre-existing constraints or witness columns"]
+pub fn capture_stage_non_fresh_inter() {
+    let input = r#"
+        namespace std::prover;
+            let capture_stage: (-> int) -> Constr[] = 9;
```

Comment:
Why `=9`? Do we need a constraint here?

Reply:
This defines a builtin. Since the builtin is "overridden" during execution and type checking, I always like to assign a value that will create an error both at type checking and during runtime when the value is used instead of being overridden.

Now that we have the prelude mechanism, we could use a fixed identifier like "std::prelude::builtin" (which is a builtin itself that always fails or something).